tests/test_packages/test_connections/test_http_client.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Peer to Peer connection and channel."""
+
+import asyncio
+import logging
+
+from fetchai.ledger.crypto import entity
+
+import pytest
+
+from aea.configurations.base import ConnectionConfig
+
+from packages.fetchai.connections.http_client.connection import HTTPClientConnection
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+class TestHTTPClient:
+    """Contains the test for the p2p connection."""
+
+    @classmethod
+    def setup_class(cls):
+        """Initialise the class."""
+        cls.address = "127.0.0.1"
+        cls.port = 8020
+        m_fet_key = "6d56fd47e98465824aa85dfe620ad3dbf092b772abc6c6a182e458b5c56ad13b"
+        cls.ent = entity.Entity.from_hex(m_fet_key)
+        cls.http_client_connection = HTTPClientConnection(
+            agent_address=cls.ent.public_key_hex,
+            provider_address=cls.address,
+            provider_port=cls.port,
+        )
+        cls.http_client_connection.loop = asyncio.get_event_loop()
+
+    async def test_initialization(self):
+        """Test the initialisation of the class."""
+        assert self.http_client_connection.agent_address == self.ent.public_key_hex
+
+    async def test_connection(self):
+        """Test the connection and disconnection from the p2p connection."""
+        await self.http_client_connection.connect()
+
+    async def test_disconnect(self):
```

Comment:
Note, you should have two separate test classes to test connect and disconnect. The reason being that you cannot guarantee for one test to run before the other as we always shuffle test execution order.

Reply:
Done